packages/runtime/container-runtime/src/gc/gcTelemetry.ts

```diff
@@ -218,86 +235,50 @@ export class GCTelemetryTracker {
 
-		// For summarizer client, queue the event so it is logged the next time GC runs if the event is still valid.
-		// For non-summarizer client, log the event now since GC won't run on it. This may result in false positives
-		// but it's a good signal nonetheless and we can consume it with a grain of salt.
-		// Inactive errors are usages of Objects that are unreferenced for at least a period of 7 days.
-		// SweepReady errors are usages of Objects that will be deleted by GC Sweep!
-		if (this.isSummarizerClient) {
-			this.pendingEventsQueue.push({
-				...unrefEventProps, // Note: Contains some properties from INodeUsageProps as well
-				usageType,
-				state,
-			});
-		} else {
-			// For non-summarizer clients, only log "Loaded" type events since these objects may not be loaded in the
-			// summarizer clients if they are based off of user actions (such as scrolling to content for these objects)
-			// Events generated:
-			// InactiveObject_Loaded, SweepReadyObject_Loaded
-			if (usageType === "Loaded") {
-				const { id, fromId, headers, gcConfigs, additionalProps, ...detailedProps } =
-					unrefEventProps;
-				const event = {
-					eventName: `${state}Object_${usageType}`,
-					...tagCodeArtifacts({ pkg: packagePath?.join("/") }),
-					stack: generateStack(),
-					id,
-					fromId,
-					headers: { ...headers },
-					details: { ...detailedProps, ...additionalProps },
-					gcConfigs,
-				};
+		if (this.isSummarizerClient || usageType === "Loaded") {
+			const event = {
+				eventName: `${state}Object_${usageType}`,
+				stack: generateStack(),
+				...loggingProps,
+			};
 
-				// These are logged as generic events and not errors because there can be false positives. The Tombstone
-				// and Delete errors are separately logged and are reliable.
-				this.mc.logger.sendTelemetryEvent(event);
-			}
+			// These are logged as generic events and not errors because there can be false positives. The Tombstone
+			// and Delete errors are separately logged and are reliable.
+			this.mc.logger.sendTelemetryEvent(event);
 		}
 	}
 
-	/**
-	 * Logs telemetry when a tombstoned object is changed, revived or loaded.
-	 */
 	private logTombstoneUsageTelemetry(
-		unrefEventProps: Omit<IUnreferencedEventProps, "state" | "usageType">,
 		nodeType: GCNodeType,
 		usageType: NodeUsageType,
-		packagePath?: readonly string[],
+		loggingProps: ITelemetryPropertiesExt,
+		allowTombstone?: boolean,
 	) {
 		// This will log the following events:
 		// GC_Tombstone_DataStore_Requested, GC_Tombstone_DataStore_Changed, GC_Tombstone_DataStore_Revived
 		// GC_Tombstone_SubDataStore_Requested, GC_Tombstone_SubDataStore_Changed, GC_Tombstone_SubDataStore_Revived
 		// GC_Tombstone_Blob_Requested, GC_Tombstone_Blob_Changed, GC_Tombstone_Blob_Revived
-		const { id, fromId, headers, gcConfigs, additionalProps, ...detailedProps } =
-			unrefEventProps;
 		const eventUsageName = usageType === "Loaded" ? "Requested" : usageType;
```

Comment:
The way you've changed this, it is going to pull a bunch of stuff that used to be in `details` up to the top level.  I'd like to leave it as-is, with key props at top level and the rest in `details`.  You can do the same destructuring trick here on `loggingProps`, to separate out the top-level stuff from the "extra" stuff to go into `details`.

I'm not sure why you removed the `IUnreferencedEventProps` type, since it gave some type safety on the props destructured here, I would have left this part totally as-is.  Maybe I'm missing some other key part of this change.

```suggestion
		const { id, fromId, headers, gcConfigs, additionalProps, ...detailedProps } =
			loggingProps;
		const eventUsageName = usageType === "Loaded" ? "Requested" : usageType;
```